### DISCUSSION TITLE
AMA thread of Jan. 2021

### DISCUSSION THREAD

**Discussion by gujun720** [milvus-io/milvus#4642] [Show and tell]
Since the Milvus community spreads all over the globe, it’s not easy for us to find a time slot to host a community event. It comes to my mind maybe we could try to do an AMA (ask me anything) on the GitHub discussion of the Milvus project. The AMA thread will be active for the whole month of Jan. 2021.  

**If you have (literally) ‘any’ question, welcome to post here.** 🥳

About the host: **Jun Gu**, Milvus project evangelist. Jun participated in the Milvus project from ever beginning. He led the effort of onboarding the Milvus project into LF AI & Data foundation as an incubation project. Jun is also the partner of Zilliz, he joined the latest round of fundraising.

**Comment by XiaoConstantine** [comment]
Great work folks!
dumping some questions I m curious about personally:
What's the proper way to get some understandings of the design decisions? 
1. Why does milvus support both HTTP and gRPC as internal communication mechanisms? why not make it gRPC only?
2. Follow up on 1), for gRPC, there's client side load balancing (rr) for example without additional infrastructure setup, is the purpose of having internal load balancers mainly for HTTP requests? will there be such complicated lb scenarios in the future?

**Reply by GuoRentong** [reply]
1. In the next version, milvus communicates via grpc, while HTTP as a wrapper (for RESTful API).
2. Load balancing in Milvus is a complicated issue, not just client or proxy matters. We will develop load balancing mechanism on query nodes and index builders, that's where the real load lies.

**Comment by 63670** [comment]
This is a very useful product！
I hope the API documentation can be more detailed in the future.

**Reply by gujun720** [reply]
Thanks for the feedback. We are aiming to improve the quality of API documentation. Which API documentation you use most?
